### DISCUSSION TITLE
How to unmount dialog/popover when closed

### DISCUSSION THREAD

**Discussion by IanVS** [ariakit/ariakit#2226] [Questions] [answered]
I have a small form that I am rendering in a popover.  When the popover opens, I want react to render the form fresh, and when it closes, if the user hasn't saved their changes, I want the form to unmount so that it can reset to the default values the next time it is opened.

From what I can tell, the children of `<Popover>` are rendered as soon as the page loads, and are not unrendered when the popover closes.  Is there a way to achieve what I'm trying to do without using `setOpen` and imperatively clearing the state?

**Comment by IanVS** [comment, marked as answer]
Duh, ignore me, I can just render the children or not depending on the state.

```ts
      <Popover state={popoverState} >
        {popoverState.open ? (
          <FilterForm
            onApply={() => {
              popoverState.hide();
            }}
          />
        ) : null}
      </Popover>
```

**Reply by diegohaz** [reply]
You can also unmount the entire `Popover` element if you want:

```jsx
popoverState.mounted && <Popover state={popoverState}>
```
